src/components/SlateEditor/plugins/visualElement/index.js

```diff
@@ -0,0 +1,46 @@
+/**
+ * Copyright (c) 2020-present, NDLA.
+ *
+ * This source code is licensed under the GPLv3 license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ */
+
+import React from 'react';
+import SlateFigure from '../embed/SlateFigure';
+export default function visualElementPlugin(options = {}) {
+  const schema = {
+    blocks: {
+      embed: {
+        isVoid: true,
+      },
+    },
+  };
+
+  /* eslint-disable react/prop-types */
+  const renderBlock = (props, editor, next) => {
+    const { node, attributes, isSelected } = props;
+    const { language } = options;
+
+    console.log(attributes);
```

Comment:
Jeg tror problemet er at props får nye verdier hver gang renderBlock kalles. Key endres både for attributes og node, og dette kjør at SlateFigure mountes på nytt. Er noen som er gode på slate (@hakmet) som vet hvorfor dette skjer? I tilsvarende createEmbedPlugin har attributes og node alltid samme key. Sammenlign http://localhost:3000/subject-matter/topic-article/22359/edit/nb og http://localhost:3000/subject-matter/learning-resource/22358/edit/nb. Har lagt inn logging på det.

Reply:
Det fungerte å endre teksten på et bilde, men ikkje for video. Kva er forskjellen mellom disse tru?

Reply:
Problemet ser ut til å være onFigureInputChange i SlateFigure som igjen kaller saveEmbedUpdates som oppdaterer noden i editoren. Eg testa å kommentere ut saveEmbedUpdates og då lukker ikkje videoen seg, men det blir heller ikkje lagra nokke.

Reply:
Kanskje EditVideo må ha knapper for Avbryt og Lagre på samme måte som bilder. Då kan kikk på disse sette verdien i noden uten at det skjer for kvart einaste tastetrykk.

Reply:
Du har nok rett der, men det er litt synd å gjøre UX endringer på grunn av en bug

Reply:
Joda, men det blir meir konsistent oppførsel mellom boksene.